content/micropython/06.communication/uart/uart.md

```diff
@@ -0,0 +1,144 @@
+author: Pedro Sousa Lima
+---
+
+Universal Asynchronous Receiver-Transmitter, or **UART**, is one of the simplest and most widely used communication protocols for connecting devices. It enables point-to-point communication over just two wires, making it a perfect fit for debugging, logging, and connecting to peripheral devices.
+
+## How UART Works
+
+UART communication relies on:
+- **TX (Transmit):** Sends data from one device to another.
+- **RX (Receive):** Receives data from the transmitting device.
+
+Unlike other protocols like I2C or SPI, UART is asynchronous, meaning it doesn’t use a clock line. Instead, the sender and receiver agree on a common data rate (baud rate), such as 9600 bits per second.
+
+Each device directly connects TX to RX and RX to TX.
+
+
+
+### Key Features of UART
+
+1. **Simple Point-to-Point Communication:** Ideal for connecting two devices.  
+2. **Flexible Speeds:** Baud rates can be adjusted to meet application needs.  
+3. **Independent Lines:** TX and RX operate independently, enabling bidirectional communication.
+
+## How to Read the Output
+
+To observe the output of your UART communication, you’ll need a secondary board (or a USB-to-UART adapter) to act as a receiver. Here’s how you can set it up:
+
+1. **Primary Board (Transmitter):** Runs the main UART code examples provided below.  
+2. **Secondary Board (Receiver):** Reads from the UART interface and displays the output on a serial monitor like Arduino IDE, PuTTY, or CoolTerm.
+
+### Circuit Setup
+- Connect the **TX1 (GPIO43)** of the primary board to the **RX** of the secondary board.  
+- Connect the **RX1 (GPIO44)** of the primary board to the **TX** of the secondary board.  
+- Connect the ground (GND) of both devices together.
+
+![How to wire](assets/UARTTestingSetup.png)
+
+### Code for Secondary Board
+The secondary board simply forwards received UART data to its serial monitor. Use the following code:
+
+```python
+from machine import UART
+
+# Initialize UART
+uart = UART(1, baudrate=9600, tx=43, rx=44)  # Adjust TX/RX pins as needed
+
+while True:
+    if uart.any():  # Check if data is available
+        ch = uart.read(1).decode()  # Read one character
+        print(ch, end="")  # Print to the console
+```
+
+### Using the Secondary Board
+1. Flash the above code to the secondary board.  
+2. Open a serial monitor for the secondary board (e.g., Arduino Lab for Micropython, Arduino IDE Serial Monitor, Putty).  
+   
+
+## Example 1: Testing UART
+
+In this first example, we’ll create a program that prints `UART says Hi!` every second, allowing you to verify that the UART connection is working.
+
+### Circuit Diagram
+For this example:
+- **Connect TX1 (GPIO43)** to the RX of your receiving device (e.g., USB-to-serial adapter).  
+- **Connect RX1 (GPIO44)** to the TX of your receiving device.  
+- **Connect GND** to the ground of the receiving device.
+
+
+### Code Example
+
+Copy the following code into your `main.py` file and run it:
```

Comment:
Does this code go on the main board? Might be good to define what board this code goes on and where the secondary board comes into the process